genie-web/src/main/java/com/netflix/genie/web/services/impl/DiskLogServiceImpl.java

```diff
@@ -0,0 +1,336 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.services.impl;
+
+import com.google.common.collect.Sets;
+import com.netflix.genie.common.internal.dto.v4.logs.JobLogFile;
+import com.netflix.genie.web.services.LogService;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.io.PathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * A local disk based implementation of the {@link LogService} interface.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+@Service
+@Slf4j
+public class DiskLogServiceImpl implements LogService {
+
+//    private static final int DEFAULT_BUFFER_SIZE = 1024;
+
+    private final Path jobsDirRoot;
+
+    /**
+     * Constructor.
+     *
+     * @param jobsDir The job directory resource
+     * @throws IOException When the job directory can't be created or isn't a directory
+     */
+    public DiskLogServiceImpl(final Resource jobsDir) throws IOException {
```

Comment:
I'm not sure I understand the difference you're trying to establish between the two. Instead of `Log` think `output` or `working directory` if you'd like. In V3 these are written directly and served directly off local disk. If we put the logs that are uploaded from somewhere else in a different location than this existing job directory space it complicates the logic to return log files in response to API requests via the V3 API.

I could see an argument for a location where running job files are kept somewhere and finished jobs are placed somewhere else but i'm not sure what the value in the distinction is here.